### PR TITLE
Check commit status when creating a manual build

### PR DESCRIPTION
**Pull request by gstamp**
A lot of CI tools use the github status API to indicate whether a given build is okay.

When doing a manual release this change queries the status and displays a warning if the build failed. It does not prevent the release, it just gives a warning.

It will look like this:

image

This could be extended to provide more information if desired but at the moment we're just keeping it reasonably simple.

### CODE REVIEW

**Comment by gstamp** [comment]
Will look into the build failure.

File: app/models/changeset.rb [hunk 1]
```diff
@@ -28,6 +28,20 @@ def commits
     @commits ||= comparison.commits.map {|data| Commit.new(repo, data) }
   end
 
+  def last_commit_status
+    if comparison.commits.nil? || comparison.commits.empty?
```
Comment (sandlerr) [inline app/models/changeset.rb]: you could use `.blank?` here http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F

File: app/models/changeset.rb [hunk 2]
```diff
@@ -28,6 +28,20 @@ def commits
...
+      StatusResult.null_result
+    else
+      ref = comparison.commits.last["sha"]
+      state = Rails.cache.fetch(status_cache_key) do
```
Comment (sandlerr) [inline app/models/changeset.rb]: let's give this an expiration time for the case when a flaky build is re-run and passes

**Comment by gstamp** [comment]
@sandlerr Done.

**Comment by sandlerr** [comment]
Great feature. Do you mind making the status check class its own file? 

Also I think the cache key doesn't need to care about the previous commit.

**Comment by gstamp** [comment]
@sandlerr Not at all. I was considering making it it's own class, it just seemed a bit small. Small things tend to grow larger though (and probably will if we add extra information from the status API).

**Comment by sandlerr** [comment]
It makes sense to me to have the Github call go inside the new class, maybe rename it to `StatusCheck` or something

**Comment by steved** [comment]
There's a CommitStatus model / controller, maybe we can combine?
https://github.com/zendesk/samson/blob/master/app/models/commit_status.rb
https://github.com/zendesk/samson/blob/master/app/controllers/commit_statuses_controller.rb

You can see it in action in your screenshot, the SHA field is outlined red.

**Comment by gstamp** [comment]
@steved Right!  Totally missed that.

I'll take a look. One difference that sticks out is that picks the status of the first status (`statuses.first.state`) rather than calling `GITHUB.combined_status()`.

**Comment by gstamp** [comment]
I'm going to close this and open a new PR.

**Comment by dasch** [comment]
Don't we already do this? The input field changes color depending on the build status.

**Comment by gstamp** [comment]
@dasch That's why I closed the PR.

**Comment by dasch** [comment]
Ok, sorry.

On Tue, Apr 7, 2015 at 10:41 AM, Glen Stampoultzis <notifications@github.com

**Pull request closed without merging**